toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -34,20 +52,44 @@ func extractPartitions(imageLoopDevice string, outDir string, basename string, p
 			case "raw":
 				// Do nothing for "raw" case.
 			case "raw-zst":
-				partitionFilepath, err = compressWithZstd(partitionFilepath)
+				partitionFilepath, err = extractRawZstPartition(partitionFilepath, skippableFrameMetadata, partitionfileName, outDir)
 				if err != nil {
 					return err
 				}
 			default:
 				return fmt.Errorf("unsupported partition format (supported: raw, raw-zst): %s", partitionFormat)
 			}
-
 			logger.Log.Infof("Partition file created: %s", partitionFilepath)
 		}
 	}
 	return nil
 }
 
+// Extract raw-zst partition
+func extractRawZstPartition(partitionRawFilepath string, skippableFrameMetadata [SkippableFramePayloadSize]byte, partitionfileName string, outDir string) (partitionFilepath string, err error) {
+	// Compress raw partition with zstd
+	tempPartitionFilepath, err := compressWithZstd(partitionRawFilepath)
+	if err != nil {
+		return "", err
+	}
+	// Create a skippable frame containing the metadata and prepend the frame to the partition file
+	partitionFilepath, err = addSkippableFrame(tempPartitionFilepath, skippableFrameMetadata, partitionfileName, outDir)
+	if err != nil {
+		return "", err
+	}
+	// Remove temp partition file
+	err = os.Remove(tempPartitionFilepath)
+	if err != nil {
+		return "", fmt.Errorf("failed to remove temp file %s:\n%w", partitionRawFilepath, err)
+	}
+	// Remove raw file since output partition format is raw-zst
+	err = os.Remove(partitionRawFilepath)
```

Comment:
You can delete `partitionRawFilepath` after `compressWithZstd()`.

Reply:
Updated